scripts/upgrade-image-from-aptly-repo.sh

```diff
@@ -0,0 +1,183 @@
+# image, by consuming a previously generated Aptly/APT repository. This
+# APT repository is the primary component of any upgrade image, and it
+# must contain a version of the delphix-entire package for each platform
+# supported, as well as all package dependencies.
+#
+# The Aptly repository will generally be created prior to calling this
+# script, by using the "aptly-repo-from-debs.sh" script.
+#
+
+. "${BASH_SOURCE%/*}/common.sh"
+
+function usage() {
+	echo "$(basename "$0"): $*" >&2
+	echo "Usage: $(basename "$0") <name>"
+	exit 2
+}
+
+function cleanup() {
+	[[ -n "$DIR" ]] && [[ -d "$DIR" ]] && rm -rf "$DIR"
+}
+
+trap cleanup EXIT
+
+[[ -n "$TOP" ]] || die "must be run inside of the git repository"
+[[ $# -gt 1 ]] && usage "too many arguments specified"
+[[ $# -lt 1 ]] && usage "too few arguments specified"
+
+#
+# One of the "inputs" to this script is an Aptly repository stored in
+# the "~/.aptly/public". Thus, we do minimal verification of this input,
+# and see if the directory exists before proceeding. The intention is
+# for this directory to have been previously generated by another
+# part of the build system; e.g. via "aptly-repo-from-debs.sh".
+#
+[[ -d ~/.aptly/public ]] || die "aptly repository not found"
+
+DIR=$(mktemp -d -t upgrade-image-from-aptly-repo.XXXXXXX)
+[[ -d "$DIR" ]] || die "unable to create temporary directory"
+pushd "$DIR" &>/dev/null || die "'pushd $DIR' failed"
+
+#
+# Generate the "payload" of the upgrade image, which will contain the
+# upgrade scripts, as well as the Aptly/APT repository. We generate this
+# "payload" tarball as a way to limit the number of files that we need
+# to checksum and store in "upgrade image" tarball itself. Rather than
+# the "upgrade image" tarball containing all of the files for the upgrade
+# scripts and APT repository, it'll only contain the "payload.tar.gz"
+# file, and a few other miscellaneous metadata files.
+#
+# We need to limit the files contained in the 'upgrade image" tarball
+# because the signing service that we use to sign the SHA256SUMS file
+# below, has a relatively small limit on the size of the file that we
+# can sign with it. Thus, we need the contents of the SHA256SUMS file to
+# be realtively small (and that file's size grows linearly with the
+# number of files contained in the "upgrade image" tarball).
+#
+tar -I pigz -cf "payload.tar.gz" \
+	-C "$TOP/upgrade/upgrade-scripts" . \
+	-C ~/.aptly/public . ||
+	die "failed to generate payload.tar.gz"
+
+cp "$TOP/upgrade/version.info.template" version.info ||
+	die "failed to copy version.info.template"
+
+VERSION=$(dpkg -f "$(find ~/.aptly/public -type f -name 'delphix-entire-*' | head -n 1)" version)
+[[ -n "$VERSION" ]] || die "failed to determine delphix-entire version"
+sed -i "s/@@VERSION@@/$VERSION/" version.info ||
+	die "failed to set VERSION in version.info file"
+
+#
+# On 6.0 versions, the virtualization application expects to find the
```

Comment:
typo: 6.0 -> 6.0+, or on version 6.0 and above

Reply:
I don't think this applies to all versions above 6.0.. IIRC, this applies strictly to 6.0.x versions, and not 6.1.x versions.. Thus, I'm more inclined to leave as is, rather than potentially imply to the reader that this comment applies to all versions above (and including) 6.0. What do you think?

Reply:
Sure, we can leave it as is.

Reply:
I verified this comment only applies to 6.0.x versions as I thought; here's the related JIRA issue: https://jira.delphix.com/browse/DLPX-69165

Reply:
Cool, thanks for posting the link, it all makes sense.